### PR TITLE
fix: refund externally voided RoundManager deals

### PR DESCRIPTION
**Pull request by Detoo**
Similar to #47 but for `RoundManager`.

- Added `reject()` and `recallEOI()` function overloads to handle the edge case when the agreement was voided outside the manager's reach
- Added related test cases

### CODE REVIEW

File: src/RoundManager.sol [hunk 1]
```diff
@@ -728,6 +730,16 @@ contract RoundManager is
...
+        reject(agreementId, true);
+    }
+
+    /// @notice Rejects an EOI and voids the deal if necessary
+    /// @dev The extra parameter allows handling the edge case where the party voided the agreement by
+    /// directly requesting to CyberAgreementRegistry and bypassing RoundManager.
+    /// In such case, we want to skip `voidContractFor()` so it does not attempt to void the contract again.
+    /// @param agreementId The agreement ID
+    /// @param isVoidAgreement True if voiding the deal
+    function reject(bytes32 agreementId, bool isVoidAgreement) public onlyOwner {
```
Comment (merisman) [inline src/RoundManager.sol:744]: should this be onlyOwnerOrSelf?

Comment (Detoo) [reply src/RoundManager.sol:744]: no it's not necessary, but there are multiple reasons behind it:

1. `reject(bytes32)` called `reject(bytes32, bool)` internally, so the `msg.sender` stays the same (the user/owner who initiated the tx)

2. Had it called with `this.reject(bytes32, bool)` instead of `reject(bytes32, bool)` then `msg.sender` would become `address(this)`
  
3. But that would still be ok because we have set RoundManager as its own owner during `deployCyberCorp()`
https://github.com/MetaLex-Tech/cybercorps-contracts/blob/caa497e7ab23faa432aeabdd246db2730972ac4c/src/CyberCorpFactory.sol#L275

**Review by merisman — APPROVED** [review]
Approved
